File: src/lib/types/section.ts
export type Severity = "high" | "medium" | "low";

export type RangeKind =
	| "context"
	| "changed-old"
	| "changed-new"
	| "added"
	| "removed";

export interface LineRange {
	file_path: string;
	start_line: number;
	end_line: number;
	kind: RangeKind;
}

export interface Concern {
	text: string;
	severity: Severity;
	file_path?: string;
	line?: number;
}

export interface ReviewSection {
	schema_version: 1;
	section_id: string;
	title: string;
	intent: string;
	files: string[];
	ranges: LineRange[];
	concerns: Concern[];
	uncovered_scenarios: Concern[];
	test_coverage_notes: string;
	base_ref: string;
	head_ref: string;
	pause_prompt: string;
}

export interface SectionMapEntry {
	section_id: string;
	title: string;
	intent: string;
}

export interface SectionMap {
	schema_version: 1;
	sections: SectionMapEntry[];
}

export type CommentKind = "inline" | "top_level";
export type CommentSide = "LEFT" | "RIGHT";

export interface CommentDraft {
	kind: CommentKind;
	body: string;
	file_path?: string;
	line?: number;
	side?: CommentSide;
}

export type CommentResultStatus = "published" | "failed";

export interface CommentResult {
	draft_id: string;
	status: CommentResultStatus;
	url?: string;
	error?: string;
}

export type SectionStatus = "pending" | "in_review" | "completed";

export interface ToolCallItem {
	tool_call_id: string;
	title: string;
	kind: string;
	status: string;
}

export type ChatItem =
	| { type: "section_map"; sections: SectionMapEntry[] }
	| { type: "review_section"; section: ReviewSection }
	| { type: "tool_call"; toolCall: ToolCallItem };

export interface ChatMessage {
	id: string;
	role: "user" | "assistant";
	text: string;
	streaming?: boolean;
	item?: ChatItem;
}

File: src-tauri/src/section.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum Severity {
    High,
    Medium,
    Low,
}

#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
#[serde(rename_all = "kebab-case")]
pub enum RangeKind {
    Context,
    ChangedOld,
    ChangedNew,
    Added,
    Removed,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LineRange {
    pub file_path: String,
    pub start_line: u32,
    pub end_line: u32,
    pub kind: RangeKind,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Concern {
    pub text: String,
    pub severity: Severity,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub file_path: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub line: Option<u32>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ReviewSection {
    pub schema_version: u32,
    pub section_id: String,
    pub title: String,
    pub intent: String,
    pub files: Vec<String>,
    pub ranges: Vec<LineRange>,
    pub concerns: Vec<Concern>,
    pub uncovered_scenarios: Vec<Concern>,
    pub test_coverage_notes: String,
    pub base_ref: String,
    pub head_ref: String,
    pub pause_prompt: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SectionMapEntry {
    pub section_id: String,
    pub title: String,
    pub intent: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SectionMap {
    pub schema_version: u32,
    pub sections: Vec<SectionMapEntry>,
}

#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum CommentKind {
    Inline,
    TopLevel,
}

#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
#[serde(rename_all = "UPPERCASE")]
pub enum CommentSide {
    Left,
    Right,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CommentDraft {
    pub kind: CommentKind,
    pub body: String,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub file_path: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub line: Option<u32>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub side: Option<CommentSide>,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum CommentResultStatus {
    Published,
    Failed,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CommentResult {
    pub draft_id: String,
    pub status: CommentResultStatus,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub url: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub error: Option<String>,
}
